tests/test_postgres_full_table_replication.py

```diff
@@ -283,247 +355,456 @@ def test_run(self):
+        )
         records_by_stream = runner.get_records_from_target_output()
+        table_version_3 = records_by_stream[test_table_name]['table_version']
+        messages = records_by_stream[test_table_name]['messages']
+
+        # verify the execpted number of records were replicated
+        self.assertEqual(4, record_count_by_stream[test_table_name])
 
-        new_table_version = records_by_stream['postgres_full_table_replication_test']['table_version']
+        # verify the message actions match expectations
+        self.assertEqual(5, len(messages))
+        self.assertEqual('upsert', messages[0]['action'])
+        self.assertEqual('upsert', messages[1]['action'])
+        self.assertEqual('upsert', messages[2]['action'])
+        self.assertEqual('upsert', messages[3]['action'])
+        self.assertEqual('activate_version', messages[4]['action'])
 
-        self.assertEqual(records_by_stream['postgres_full_table_replication_test']['messages'][0]['action'],
-                         'upsert')
+        # verify the new table version increased on the second sync
+        self.assertGreater(table_version_3, table_version_2)
 
-        self.assertEqual(records_by_stream['postgres_full_table_replication_test']['messages'][1]['action'],
-                         'upsert')
+        # verify the persisted schema still matches expectations
+        self.assertEqual(expected_schemas[test_table_name], records_by_stream[test_table_name]['schema'])
 
-        self.assertEqual(records_by_stream['postgres_full_table_replication_test']['messages'][2]['action'],
-                         'upsert')
 
-        self.assertEqual(records_by_stream['postgres_full_table_replication_test']['messages'][3]['action'],
-                         'activate_version')
+        # NB | This is a little tough to track mentally so here's a breakdown of
+        #      the order of operations by expected records indexes:
 
-        new_table_version = records_by_stream['postgres_full_table_replication_test']['table_version']
+        #      Prior to Sync 1
+        #        insert 0, 1, 2
 
-        self.assertGreater(new_table_version, table_version,
-                           msg="table version {} didn't increate from {} on the second run".format(new_table_version, table_version))
+        #      Prior to Sync 2
+        #        No db changes
 
-        # verifications about individual records
-        for stream, recs in records_by_stream.items():
-            # verify the persisted schema was correct
-            self.assertEqual(recs['schema'],
-                             expected_schemas[stream],
-                             msg="Persisted schema did not match expected schema for stream `{}`.".format(stream))
+        #      Prior to Sync 3
+        #        insert 3, 4, 5
+        #        update 0, 4
+        #        delete 1, 5
 
+        #      Resulting Synced Records: 2, 3, 0, 4
```

Comment:
I believe the full table select will sort based on `xmin` in ascending order. So we will replicate in order of primary key value on the initial sync here since we only have done inserts on the table and all row versions are the same therefore all xmins are the same. I will review with devs and product to confirm this is the expected behavior.

Reply:
So I think if I were to only update expected record 0 and not 4, the order would be 2, 3, 4, 0. Since only 0 will have xmin greater than 1 (or whatever the unit is that postgres does).

Reply:
Confirmed with Dan we do not order by pk at all here. It is always by xmin.

Reply:
My initial explanation was incorrect though. The ordering uses xmin only. So the fact that the first sync returns records in order of pk was just coincidental. If I had inserted records with pk values out of order such as: 0, 2, 1 the first sync would return them in that order since xmin will increment for every insert.